runconfig/parse.go

```diff
@@ -513,3 +505,31 @@ func ParseDevice(device string) (DeviceMapping, error) {
 	}
 	return deviceMapping, nil
 }
+
+// ParseVolumesIntoBinds parses any volumes passed by the client and moves any
+// volumes which are in Config.Volumes into HostConfig.Binds.
+func ParseVolumesIntoBinds(c *Config, hc *HostConfig) (*Config, *HostConfig, error) {
```

Comment:
This processing on the daemon side is really an API change since there was not processing before the path could include anyting (including `:`) and be totally legit.